scripts/fast-reboot

```diff
@@ -133,27 +179,25 @@ fi
 # Kill swss dockers
 docker kill swss
 
-# syncd service stop is capable of handling both warm/fast/cold shutdown
-systemctl stop syncd
-
-# Warm reboot: dump state to host disk
-# Note: even if syncd changed ASIC_DB before killed, we don't care
+# Pre-shtudown syncd and Stop teamd gracefully
 if [[ "$REBOOT_TYPE" = "warm-reboot" ]]; then
-    # Dump redis content to a file 'dump.rdb' in warmboot directory
-    mkdir -p $WARM_DIR
-    # Delete keys in stateDB except FDB_TABLE|* and WARM_RESTA*
-    redis-cli -n 6 eval "
-        for _, k in ipairs(redis.call('keys', '*')) do
-            if not string.match(k, 'FDB_TABLE|') and not string.match(k, 'WARM_RESTART_TABLE|') then
-                redis.call('del', k)
-            end
-        end
-    " 0
-    redis-cli save
-    docker cp database:/var/lib/redis/$REDIS_FILE $WARM_DIR
-    docker exec -i database rm /var/lib/redis/$REDIS_FILE
+    initialize_pre_shutdown
+
+    request_pre_shutdown
+
+    wait_for_pre_shutdown_complete_or_fail
+
+    backup_datebase
+
+    # Send USR1 signal to all teamd instances to stop them
+    # It will prepare teamd for warm-reboot
+    # Note: We must send USR1 signal before syncd, because it will send the last packet through CPU port
+    docker exec -i teamd pkill -USR1 teamd > /dev/null
```

Comment:
We are trying to minimize the teamd down time. But since now DB backup is only a second. We might be able to backup DB after killing teamd. Any objections?

Reply:
I think we have an agreement that this PR can move forward with following changes coming as separate PR(s):

- Drain the event pipe before pre-shutdown.
- Stop handling any pending events after pre-shutdown except shutdown request.

Please confirm.